### ISSUE TITLE
Plans to support permessage-deflate?

### ISSUE THREAD

**Issue by quinnj** [citkane/SimpleWebsockets.jl#6]
Hey, great package! I feel silly not knowing this existed, as I recently tried to look over various websocket packages in Julia to see if we needed to keep the half-working implementation in HTTP.jl or if I needed to overhaul. If I would have found this, I think we could have directed folks here. In any case, I ended up [overhauling](https://github.com/JuliaWeb/HTTP.jl/pull/843) the code, including doing integrated testing against the autobahn testsuite.

Anyway, I'm going around finding HTTP.jl dependents who may need help updating to 1.0 and that's how I found the package here.

Long story short, I'm now curious if there are plans to support the `permessage-deflate` extension to the websocket protocol? We had [this issue](https://github.com/JuliaWeb/HTTP.jl/issues/853) opened up for HTTP.jl recently and I'm just wondering if people have ideas on the best way to support.

Cheers!

**Comment by citkane** [comment]
Hi quinnj, thanks!

In short - No immediate plans to support `permessage-deflate`.
I have however left `Sec-WebSocket-Extensions` hooks for the client [here](https://github.com/citkane/SimpleWebsockets.jl/blob/aca8da34faac183e20e960ab74e1b9bb2adfce6e/src/lib/WebsocketClient.jl#L108) and [here](https://github.com/citkane/SimpleWebsockets.jl/blob/aca8da34faac183e20e960ab74e1b9bb2adfce6e/src/lib/WebsocketClient.jl#L137) and the server [here](https://github.com/citkane/SimpleWebsockets.jl/blob/aca8da34faac183e20e960ab74e1b9bb2adfce6e/src/lib/WebsocketServer.jl#L185).

This is partly because I have not yet implemented `Sec-WebSocket-Extensions` and need to take time to understand the working innards of `permessage-deflate`. ATM I cannot afford to do this, and the package does for me what I need it to do.

This project is pretty much an architectural translation of NODEjs [websocket](https://www.npmjs.com/package/websocket) into Julia. The API interface with event hooks, error handlers, auth functions and `emit` and `broadcast` server methods is a very powerful toolkit for building higher level application logic.

I started development here by trying to use `HTTP.Websockets` as a dependency, and then wrap my API logic around this, but found that the code did not give me the handles I needed out of the box. It would be nice if all the low level WebSocket stuff was exposed, documented and maintained as a standards compliant library in `HTTP.WebSockets`. This could enable any number of Julia WebSocket API libraries (such as this one) to evolve around (and contribute to) a common solid foundation.

Anyway - I have no idea how you are fixed. 
I am not going to be in a position to do timely maintenance and admin on this repository, but I am happy to invite you  as a collaborator if you wish to work things in parallel.

All the best,
M
